src/backend/looped/weapons/cage_gun.cpp

```diff
@@ -7,23 +7,49 @@ namespace big
 {
 	void looped::weapons_cage_gun()
 	{
-		if (const bool bCageGun = g.weapons.custom_weapon == CustomWeapon::CAGE_GUN; bCageGun)
+		if (g.self.custom_weapon_stop)
 		{
-			if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_AIM))
+			if (const bool bCageGun = g.weapons.custom_weapon == CustomWeapon::CAGE_GUN; bCageGun && WEAPON::IS_PED_ARMED(self::ped, 4 | 2))
```

Comment:
I hadn't really taken a look but this code can be written differently, there's a lot of code duplication for no real reason.

If you write this if statement differently you don't need to add the else statement with a whole copy of the original code.

```cpp
if (const bool bCageGun = g.weapons.custom_weapon == CustomWeapon::CAGE_GUN; bCageGun && (!g.self.custom_weapon_stop || WEAPON::IS_PED_ARMED(self::ped, 4 | 2)))
```

Writing the if statement like this will skip the WEAPON::IS_PED_ARMED call if `g.self.custom_weapon_stop` is false and if `g.self.custom_weapon_stop` is true then it will call WEAPON::IS_PED_ARMED function.

Please implement it like that for all custom weapons.